src/contexts/AccountsContext.tsx

```diff
@@ -49,6 +51,7 @@ export const useAccounts = () => useContext(AccountsContext);
 export const AccountsContextProvider = ({ children }: AccountsContextProviderProps) => {
   const [availableAccounts, setAvailableAccounts] = useState<Account[] | null>(null);
   const [activeAccount, setActiveAccount] = useState<Account | null>(null);
+  const [activeChain, setActiveChain] = useState<Chain | null>(null);
```

Comment:
we have storedChain, setStoredChain
why do we need this?

Reply:
removed